apps/src/dance/ai/utils.ts

```diff
@@ -12,18 +13,50 @@ export const generateBlocksFromResult = (
   const blocksSvg = generateBlocks(workspace);
 
   // Foreground block.
-  blocksSvg[0].setFieldValue(effect.foregroundEffect, 'EFFECT');
+  validateAndSetFieldValue(
+    blocksSvg[0].getField('EFFECT') as FieldDropdown,
+    effect.foregroundEffect,
+    FieldKey.FOREGROUND_EFFECT
+  );
 
   // Background block.
-  blocksSvg[1].setFieldValue(effect.backgroundEffect, 'EFFECT');
-  blocksSvg[1].setFieldValue(effect.backgroundColor, 'PALETTE');
+  validateAndSetFieldValue(
+    blocksSvg[1].getField('EFFECT') as FieldDropdown,
+    effect.backgroundEffect,
+    FieldKey.BACKGROUND_EFFECT
+  );
+  validateAndSetFieldValue(
+    blocksSvg[1].getField('PALETTE') as FieldDropdown,
+    effect.backgroundColor,
+    FieldKey.BACKGROUND_PALETTE
+  );
 
   // Connect the blocks.
   blocksSvg[0].nextConnection.connect(blocksSvg[1].previousConnection);
 
   return blocksSvg;
 };
 
+function validateAndSetFieldValue(
+  dropdown: FieldDropdown,
+  value: string,
```

Comment:
minor question/suggestion...is the `value` here always an `effect` (either foreground or background?)

If so, I'd suggest changing this variable to `effect` for clarity. If it's more generic and we could expect non-effect values, then leave as is.

Reply:
Hmm it can be either a background effect, foreground effect, or a background color. I think in the context of this function specifically, I went with `value` since this by itself this is just a generic function that checks if a given value is present in the list of dropdown options for any dropdown field. I think just to avoid confusion between background/foreground effect and color, I may leave it as is?